src/d3/hapticplot.d3.ts

```diff
@@ -0,0 +1,137 @@
+// import * as _d3 from 'd3';
+
+// declare global {
+// const d3: typeof _d3;
+// }
+
+// for running on unit tests
+import * as d3 from 'd3';
+import * as THREE from 'three';
+
+export class Hapticplot{
+    private data: number[];
+    private shape: string;
+    private container: HTMLElement | null;
+    private hscale: d3.ScaleLinear<number, number>;
+
+  constructor(shape: string) {
+    this.shape = shape;
+  }
+  init(container: HTMLElement | null, data: number[]){
+    this.data = data;
+    this.container = container;
+    // create a scale so that there is correspondence between data set and screen render
+    this.hscale = d3.scaleLinear();
+    this.hscale.domain([0, d3.max(this.data) as number])       // max of dataset
+    .range([0, 100]);                                      // linear mapping of data set values to values from 0 to 10
+    this.createSky();
+    this.generatePts();
+    this.setColor('blue');
+    this.setRadius(0.1);
+    this.setInteraction();
+    this.addEventListeners();
+    this.createGridPlane();
+  }
+
+  private generatePts() {
+     /*
+     Generates points based on initilization data
+      - "enter" identifies any DOM elements to be added when # array elements & # DOM elements don't match
+      - select all generated entities and updates their position based on ingested data
+    */
+    d3.select(this.container).selectAll(this.shape).data(this.data).enter().append(this.shape);
+    d3.select(this.container).selectAll(this.shape).attr('position', (d, i) => {
+      const x = i/10;
+      const y = this.data[i];
+      const z = -1;
+      return `${x} ${y} ${z}`;
+    });
+  }
+  private setColor(color) {
+    /*
+    Adds given color property to all generated entities
+    */
+    d3.select(this.container).selectAll(this.shape).attr('color', () => {
+      return color;
+    });
+  }
+
+  private setRadius(size) {
+    /*
+    Sets the radius of generated entities
+    */
+    d3.select(this.container).selectAll(this.shape).attr('radius', size);
+  }
+
+  private setInteraction() {
+    /*
+    adds interaction properties to all objects of type this.shape
+      - enables controller interaction with tagged entities
+    */
+    d3.select(this.container).selectAll(this.shape)
+      .attr('hoverable', '')
+      .attr('grabbable', '')
+      .attr('stretchable', '')
+      .attr('draggable', '')
+      .attr('dropppable', '');
+
+  }
+
+  private addEventListeners() {
+    /*
+      adds event listeners to react if a generated data object is hovered or un-hovered
+    */
+    d3.select(this.container).selectAll(this.shape).on('stateadded', function(d, i) {
+      if (d3.event.detail === 'hovered'){
+        d3.select(this).attr('color', 'orange');
+      }
+    });
+    d3.select(this.container).selectAll(this.shape).on('stateremoved', function(d, i) {
+      if (d3.event.detail === 'hovered'){
+        d3.select(this).attr('color', 'blue');
+      }
+    });
+  }
+
+
+
+  private createSky(){
```

Comment:
It looks like createSky and createGridPlane are both repeated in hapticplot and scatterplot; can you DRY this up?  Maybe a helper class that takes care of creating these things and can be used by both.

Reply:
Good call! I'm planning on deviating my implementation of createGridPlanes significantly, so I think it will be best to keep that one seperate. createSky is pretty generic though, I'll work on creating a helper class and talk with carlos about whether what else we can do to avoid redundancies.